### PR TITLE
Add a secondary headless backend

### PR DESCRIPTION
**Pull request by emersion**
This allows the create_output command to work on DRM too.

Depends on ~~https://github.com/swaywm/wlroots/pull/2111~~, ~~https://github.com/swaywm/sway/pull/5215~~ and ~~https://github.com/swaywm/wlroots/pull/2063~~

cc @any1

### CODE REVIEW

File: sway/server.c [hunk 1]
```diff
@@ -153,9 +154,15 @@ bool server_init(struct sway_server *server) {
 		return false;
 	}
 
+	server->noop_backend = wlr_noop_backend_create(server->wl_display);
+
 	struct wlr_output *wlr_output = wlr_noop_add_output(server->noop_backend);
 	root->noop_output = output_create(wlr_output);
 
+	server->headless_backend =
```
Comment (any1) [inline sway/server.c:162]: What happens here if a headless backend was already created in `wlr_backend_autocreate`?

Comment (emersion) [reply sway/server.c:162]: Then we get two headless backends, but it's no big deal, since they share the renderer.

Comment (any1) [reply sway/server.c:162]: Might this be a potential source of confusion for future contributors?

Comment (emersion) [reply sway/server.c:162]: I think conditionally creating the headless backend would be even more confusing.

**Comment by ddevault** [comment]
Rebase?

File: sway/server.c (hunk 1 above)
Comment (emersion) [reply sway/server.c:162]: Added a comment to disambiguate

**Comment by emersion** [comment]
Rebased again

**Comment by ddevault** [comment]
Thanks!

**Pull request merged**